In [16]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sns
%matplotlib inline

# For this homework is used laptops dataset

In [6]:
df=pd.read_csv('./data/2024_laptops.csv')

In [7]:
df

Laptop       Status   Brand  \
0     ASUS ExpertBook B1 B1502CBA-EJ0436X Intel Core...          New    Asus   
1     Alurin Go Start Intel Celeron N4020/8GB/256GB ...          New  Alurin   
2     ASUS ExpertBook B1 B1502CBA-EJ0424X Intel Core...          New    Asus   
3     MSI Katana GF66 12UC-082XES Intel Core i7-1270...          New     MSI   
4     HP 15S-FQ5085NS Intel Core i5-1235U/16GB/512GB...          New      HP   
...                                                 ...          ...     ...   
2155  Razer Blade 17 FHD 360Hz Intel Core i7-11800H/...  Refurbished   Razer   
2156  Razer Blade 17 FHD 360Hz Intel Core i7-11800H/...  Refurbished   Razer   
2157  Razer Blade 17 FHD 360Hz Intel Core i7-11800H/...  Refurbished   Razer   
2158  Razer Book 13 Intel Evo Core i7-1165G7/16GB/1T...  Refurbished   Razer   
2159  Razer Book FHD+ Intel Evo Core i7-1165G7/16GB/...  Refurbished   Razer   

           Model                CPU  RAM  Storage Storage type       GPU  \
0     ExpertBook      Intel Core i5    8      512          SSD       NaN   
1             Go      Intel Celeron    8      256          SSD       NaN   
2     ExpertBook      Intel Core i3    8      256          SSD       NaN   
3         Katana      Intel Core i7   16     1000          SSD  RTX 3050   
4            15S      Intel Core i5   16      512          SSD       NaN   
...          ...                ...  ...      ...          ...       ...   
2155       Blade      Intel Core i7   16     1000          SSD  RTX 3060   
2156       Blade      Intel Core i7   16     1000          SSD  RTX 3070   
2157       Blade      Intel Core i7   32     1000          SSD  RTX 3080   
2158        Book  Intel Evo Core i7   16     1000          SSD       NaN   
2159        Book  Intel Evo Core i7   16      256          SSD       NaN   

      Screen Touch  Final Price  
0       15.6    No      1009.00  
1       15.6    No       299.00  
2       15.6    No       789.00  
3       15.6    No      1199.00  
4       15.6    No       669.01  
...      ...   ...          ...  
2155    17.3    No      2699.99  
2156    17.3    No      2899.99  
2157    17.3    No      3399.99  
2158    13.4   Yes      1899.99  
2159    13.4   Yes      1699.99  

[2160 rows x 12 columns]

In [38]:
df.dtypes

Laptop           object
Status           object
Brand            object
Model            object
CPU              object
RAM               int64
Storage           int64
Storage type     object
GPU              object
Screen          float64
Touch            object
Final Price     float64
dtype: object

# Q1. Pandas version

What's the version of Pandas that you installed?

In [39]:
pd.__version__

'2.3.1'

# Q2. Records count

How many records are in the dataset?

- 12
- 1000
- 2160
- 12160

In [9]:
len(df)

2160

# Q3. Laptop brands

How many laptop brands are presented in the dataset?

- 12
- 27
- 28
- 2160

In [14]:
len(df.Brand.unique())

27

# Q4. Missing values

How many columns in the dataset have missing values?

- 0
- 1
- 2
- 3

# Q5. Maximum final price

What's the maximum final price of Dell notebooks in the dataset?

- 869
- 3691
- 3849
- 3936

In [37]:
df[df.Brand == 'Dell']['Final Price'].max()

np.float64(3936.0)

# Q6. Median value of Screen

Find the median value of Screen column in the dataset.
Next, calculate the most frequent value of the same Screen column.
Use fillna method to fill the missing values in Screen column with the most frequent value from the previous step.
Now, calculate the median value of Screen once again.
Has it changed?

Hint: refer to existing mode and median functions to complete the task.

- Yes
- No

# Q7. Sum of weights

Select all the "Innjoo" laptops from the dataset.
Select only columns RAM, Storage, Screen.
Get the underlying NumPy array. Let's call it X.
Compute matrix-matrix multiplication between the transpose of X and X. To get the transpose, use X.T. Let's call the result XTX.
Compute the inverse of XTX.
Create an array y with values [1100, 1300, 800, 900, 1000, 1100].
Multiply the inverse of XTX with the transpose of X, and then multiply the result by y. Call the result w.
What's the sum of all the elements of the result?
Note: You just implemented linear regression. We'll talk about it in the next lesson.

- 0.43
- 45.29
- 45.58
- 91.30